src/Network/Transport/TCP.hs

```diff
@@ -993,21 +980,21 @@ handleIncomingMessages (ourEndPoint, theirEndPoint) = do
     -- exception thrown by 'recv'.
     go :: N.Socket -> IO ()
     go sock = do
-      lcid <- recvInt32 sock :: IO LightweightConnectionId
+      lcid <- recvWord32 sock :: IO Word32
```

Comment:
Isn't it ok to keep the `:: IO LightweightConnectionId` ? Isn't `LightweightConnectionId` a Word32?